File: conformance/bundles/03-expected-failure/DivisionSteps.java
package dev.varar.conformance.bundle03;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;

/** Java sibling of {@code division.steps.ts} / {@code division.steps.py} (bundle {@code 03-expected-failure}). */
public final class DivisionSteps implements StepDefinitions<DivisionSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.defineState(Ctx::new);

        s.stimulus(
                "I divide {int} by {int}",
                (Ctx ctx, Integer a, Integer b) -> {
                    if (b == 0) throw new ArithmeticException("division by zero");
                    return ctx;
                });
    }
}

File: conformance/bundles/03-expected-failure/division.steps.kt
@file:JvmName("DivisionSteps")

// Kotlin sibling of division.steps.ts / division.steps.py / DivisionSteps.java
// (bundle 03-expected-failure).
package dev.varar.kotlin.conformance.bundle03

import dev.varar.kotlin.stimulus
import dev.varar.kotlin.steps

class Ctx

val steps = steps(::Ctx) {
    stimulus("I divide {int} by {int}") { a: Int, b: Int ->
        if (b == 0) throw ArithmeticException("division by zero")
        this
    }
}
